Port Parkstation Benches

rosieposieeee opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->

![image](https://github.com/DeltaV-Station/Delta-v-rebase/assets/52761126/fecd4e5e-b4ce-4417-99d8-6a89ee453abb)

Motha freakin benches baby.

Comes with all the parkstation ones that are also on frontier.

The corner sofa doesn't buckle but otherwise they're functional. Should spice up mapping a bunch.

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- add: Added benches, sofas, and pews!

DEATHB4DEFEAT commented on Content.Server/Speech/EntitySystems/GermanAccentSystem.cs:
```diff
@@ -0,0 +1,54 @@
+        //[Dependency] private readonly IRobustRandom _random = default!;
+        private const string GermanKey = "German";
+
+        private static readonly IReadOnlyDictionary<string, string> SpecialWords = new Dictionary<string, string>()
+        {
+            { "yeah",       "ja"                },
+            { "yes",        "ja"                },
+            { "thank you",  "danke"             },
+            { "thanks",     "danke"             },
+            { "shit",       "sheiße"            },
+            { "damned",     "verdammt"          },
+            { "fucking",    "verdammt"          },
+            { "fucked",     "gefickt"           },
+            { "hello",      "guten tag"         },
+            { "bye",        "tschüss"           },
+            { "goodbye",    "auf viedersehen"   },
+            { "hell",       "hölle"             },
+            { "please",     "bitte"             },
+            { "true",       "stimmt"            },
+            { "my god",     "mein gott"         },
+        };
```
I believe there's a system for doing this which uses locale

moonsky70 replied:
I'll look into this! This system is based off of the skeleton's accent system, which replaces words in a pretty similar way. I didn't see locale used anywhere in the existing accent components, but I will go ahead and search that down <3

moonsky70 replied:
Any recollection of where I might find the locale-based system? I'm not finding anything in my solution or on the discords, but that doesn't mean it isn't there.

DEATHB4DEFEAT commented on Content.Server/Traits/Assorted/AlcoholismSystem.cs:
```diff
@@ -0,0 +1,65 @@
+
+        if (!_timing.IsFirstTimePredicted)
+            return;
+        foreach(var player in _player.GetAllPlayerData())
```
use a query for the component

moonsky70 replied:
I'm assuming this is AllEntityQuery? I had no idea this was a thing, thank you! I'll go ahead and switch this over. I'm assuming that this will query entities other than players, which would make it slightly more expensive but also allow it to be used universally, and that sounds good.

DEATHB4DEFEAT replied:
There's a function for component queries, EntitySystem's EntityQueryEnumerator

moonsky70 replied:
What's the functional difference between AllEntityQuery and simply EntityQuery? Neither of them have code summaries, though I spose I could go check how they were written to try n figure it out

DEATHB4DEFEAT commented on Content.Server/Traits/Assorted/AlcoholismSystem.cs:
this should be made into a generic addiction system

moonsky70 replied:
Are you thinking just changing this to apply to any chemicals in the chemstream, rather than just ethanol? Unfortunately, while we discussed adding tobacco/marijuana dependency, currently cigarettes are purely cosmetic and don't add to the chemstream.

DEATHB4DEFEAT replied:
I think cigarettes do add to the chemstream, the effects are just really weak and the amount of reagent is so low it gets processed immediately. I don't know how else the Syndicate Smokes would work.

TadJohnson00 replied:
Syndicate cigarettes and other smokables with chems injected into them can be used to metabolise reagents, but nicotine itself does not exist as a reagent yet, and thus most cigarettes are useless. Nonetheless, a generic addiction system would certainly open possibilities for nicotine, desoxyephedrine, and perhaps other chemicals to pose a minor long-term liability to the station crew and medical department.

DEATHB4DEFEAT commented on Content.Shared/Traits/Assorted/AlcoholicComponent.cs:
```diff
@@ -0,0 +1,75 @@
+}
+
+[Serializable, NetSerializable]
+public sealed class AlcoholicComponentState : ComponentState
```
use the automatic state generation tag on the component instead of this and a system
I dont remember the exact name, I believe PDA uses it if you want an example

DEATHB4DEFEAT commented on Content.Client/Traits/AlcoholismSystem.cs:
why's this here

moonsky70 replied:
You know? I'm not sure. I'm still relatively unused to the component system and copied this framework over from a different trait... But alcoholism isn't processed at all clientside, so I don't really need it. Will remove xP

DebugOk reviewed (changes requested):
Cant merge due to conflicts and requested changes

DebugOk commented:
Closing as stale

rosieposieeee committed:
bench time baby

rosieposieeee committed:
goodbye

DebugOk reviewed (dismissed):
SimpleStation14 would be yet another addition namespace that has to be watched. I'm going to need to think on that first before looking into this.

Marked as requested changes to blocked merging

DEATHB4DEFEAT reviewed (changes requested):
Psp was lazy with the sources

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/parkbench_bamboo.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/parkbench_bamboo.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/parkbench_wooden.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/parkbench_wooden.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/pews.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/sofa_corp.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/steel_bench.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/steel_bench.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/steel_bench_white.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

rosieposieeee committed:
Update Resources/Textures/SimpleStation14/Structures/Furniture/Benches/steel_bench_white.rsi/meta.json

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: rosieposie <52761126+rosieposieeee@users.noreply.github.com>

VMSolidus committed:
Vulpakin Rebalance (#69)

* Branch contents

* Update attributions.yml

* Update speech_emote_sounds.yml

Commit:
Automatic Changelog Update (#69)

FluffiestFloof committed:
Prevent natural existence of the vape (#79)

* prevent natural existence of the vape

* nospawn the vape

Commit:
Automatic Changelog Update (#79)

FluffiestFloof committed:
Praying Mantis (#86)

* Mantis mail for detective

Forensic Pads are no longer for Mantis, as the detective will take over the investigating crimes job.

* Mantis

* Remove forensic and pill

* Doesn't need this headset

* buh

* Fix capitalization

I swear I have a brain

* Minor spelling mistake, PR closed

* bread

* Rename to psionic mantis + antag advantage

* Zipties and mindbreaker pills

Mindbreaker doesn't do anything yet

* Update job-description.ftl

minor spelling mistake

Signed-off-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>

---------

Signed-off-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>
Co-authored-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>

Commit:
Automatic Changelog Update (#86)

Colin-Tel committed:
Adds PDA boxes and adjusts PTech stock (#80)

* Adds PDA boxes and adjusts PTech stock

Adds six different PDA boxes and places them in appropriate lockers. Also adjusts the PTech inventory so it makes more sense with what we're going for. Also comments out the service headsets in the PietyVend.

* Update pda.yml

lowercase-ified the boxes

* Update pda.yml

oopsies forgot the mantis

* Update pda.yml

Commit:
Automatic Changelog Update (#80)

DebugOk committed:
Sync master up to 1/10 (#74)

* Automatic changelog update

* Make NPC names proper nouns & fix some genders (#20534)

* Proper names & genders

* Uppercase proper names

* Make Smile female

* fix bingus wrinkly head (#20531)

* rouny meat and steak (#20526)

* lost friendship flavor

* add rouny steak

* rouny special meat

* rouny meat textures

---------

Co-authored-by: deltanedas <@deltanedas:kde.org>

* Automatic changelog update

* Wearable Wet Floor Sign and 'Janitorial Suicide Vest' (#20311)

* Explosive wet floor sign & janitorial suicide vest

* fix attributions

* Remove name & desc from explosive wet floor sign

* Make wet floor sign chameleonable

* Automatic changelog update

* Minor slippery stuff (#20535)

* Update submodule to 162.2.0 (#20570)

* Predicted armor (#20560)

* clean up some lines in smile the slime prototype (#20552)

* Revert "Use full file path for temp replays (#19002)" (#20545)

* Add EyesGlasses into ClothesMate (#20523)

* Automatic changelog update

* Fix Punpun crew monitor sensor (#20484)

* Automatic changelog update

* EasyPry airlocks for arrivals. Now also prying refactor I guess (#19394)

Co-authored-by: metalgearsloth <comedian_vs_clown@hotmail.com>

* Automatic changelog update

* Make arcades hackable again (#20555)

* Automatic changelog update

* Health alert tweaks (#20557)

* Automatic changelog update

* Add active and ended game rule components, generic TryRoundStartAttempt and minPlayers field (#20564)

* Improve active game rule querying, add generic try round start attempt method, move minPlayers to GameRuleComponent

* Nukeops todo and cleanup

* Remove Active field

* Add EndedGameRuleComponent

* bartender suit (#20521)

* give me a drink bartender

* guh

* aARG

* Necropolis and mine walls (#20578)

* Automatic changelog update

* archaic accent tweaks (#20567)

* Automatic changelog update

* Add confirmation to kick and respawn in the admin player actions panel (#20542)

* Remove windows-latest CI runs, remove unused build-test-release.yml (#20540)

* Change .editorconfig to keep existing attribute arrangement (#20538)

* Automatic changelog update

* Catch replay start and end errors on round restarts (#20565)

* Update trivial components to use auto comp states (#20539)

* Fix role unbans not applying in real time (#20547)

* Disable AHelp buttons when no player is selected, update button styling (#20568)

* Automatic changelog update

* Added a toggle fullscreen button (default F11) (#20272)

* Added a toggle fullscreen button (default F11)

* Removed un-needed comments

* Review Requested Changes

* Fixed Acidental Spacing Change

* bwoink, removed extraneous code

* nothing, litterally

* Automatic changelog update

* Fix not networking markings (#20588)

* Automatic changelog update

* Update RobustToolbox to v162.2.1 (#20590)

* Automatic changelog update

* Glorfcode (force say on damage/stun/crit) (#20562)

* Automatic changelog update

* Fix AHelp progressively showing more AHelp panels (#20591)

* Automatic changelog update

* caninsert entitystorage tweaks (#20589)

* Automatic changelog update

* Fix mapping actions crashing on load (#20592)

* Fix loading a map with a ThirstComponent crashing the game (#20594)

* Fix electrocution displaying real name of disguised players (#20603)

* Organ fixes (#20488)

* Automatic changelog update

* Kettle medical rework (#20435)

* Initial try at medical rework for kettle

* Re-add decal decorations to Kettle's medical

* Add more lockers to kettle's med

* Actually upload kettle's map, not its proto

* Automatic changelog update

* Added blocked visuals to volumetric pump (#20610)

* Automatic changelog update

* Move TimedDespawn to engine (#20515)

* Update submodule to 163.0.0 (#20616)

* Remove v0.1 version number from local main menu screen (#20617)

* Remove cloneData parameter from AutoNetworkedField (#20596)

* Update submodule to 164.0.0 (#20618)

* fix cognizine ghost role (#20632)

Co-authored-by: deltanedas <@deltanedas:kde.org>

* Automatic changelog update

* Move ID layer one pixel to correct alignment (#20630)

* Update nukie hardsuit descriptions (#20529)

* Make holofans destructable (#20445)

* Automatic changelog update

* bowl is open (#20453)

Co-authored-by: deltanedas <@deltanedas:kde.org>

* Add TestPair.WaitCommand() (#20615)

* Rename ThreatPrototype and mark fields as required (#20611)

* Adjust hard bomb shape (#20608)

* dragon refactor, objectives and use GenericAntag (#20201)

Co-authored-by: deltanedas <@deltanedas:kde.org>

* Automatic changelog update

* Added generic empty liquids tank (#20563)

* Move view variables verb to the top of the list with no category and localize it (#20546)

* saltern update (#20325)

Co-authored-by: deltanedas <@deltanedas:kde.org>

* Fix followers leaking (#20643)

* Add Winter Boots (#20622)

* add

* fix prototype

* Add QM mantle (#20621)

* add

* fix prototype

* Automatic changelog update

* Slime mobs breathe nitrogen and resprite their organs (#20577)

* Slimes breathe nitrogen and resprite their organs

* ups

* mmm

* Automatic changelog update

* Fix the new lizard horn's consistency + issue (#20620)

* fix

* add

* Revert "add"

This reverts commit a054a3204a8f185a94ceb80b1bd3bc9f30423711.

* Add RandomHumanoidAppearance component to for space ninjas (#20605)

* Automatic changelog update

* Space cat breathes space (#20550)

* Space cat breathes space

Made Space Cat lungs ROBUST

* Made Space Cat's lungs ROBUST 2.0

* Automatic changelog update

* Wide anomaly locator (#20581)

* Add files via upload

* Add files via upload

* Add files via upload

* Add files via upload

* add textures

* fix encoding

* fix 2

* Automatic changelog update

* Fixed Telescopic Shield Lighting (#20650)

* Fixed Telescopic Shield Lighting Bug fix #20199

* no need for these at all

---------

Co-authored-by: Nemanja <98561806+EmoGarbage404@users.noreply.github.com>

* Automatic changelog update

* Add a special hardsuit for nukie medic (#20471)

* Automatic changelog update

* Carry over other mutations when doing species mutation (#20551)

* Automatic changelog update

* Add Spanish accent to poncho and sombrero (#20377)

---------

Co-authored-by: PJBot <pieterjan.briers+bot@gmail.com>
Co-authored-by: Psychpsyo <60073468+Psychpsyo@users.noreply.github.com>
Co-authored-by: deltanedas <39013340+deltanedas@users.noreply.github.com>
Co-authored-by: metalgearsloth <31366439+metalgearsloth@users.noreply.github.com>
Co-authored-by: Nemanja <98561806+EmoGarbage404@users.noreply.github.com>
Co-authored-by: ravage <142820619+ravage123321@users.noreply.github.com>
Co-authored-by: Pieter-Jan Briers <pieterjan.briers+git@gmail.com>
Co-authored-by: LEVELcat <68501903+LEVELcat@users.noreply.github.com>
Co-authored-by: Repo <47093363+Titian3@users.noreply.github.com>
Co-authored-by: nikthechampiongr <32041239+nikthechampiongr@users.noreply.github.com>
Co-authored-by: metalgearsloth <comedian_vs_clown@hotmail.com>
Co-authored-by: Doru991 <75124791+Doru991@users.noreply.github.com>
Co-authored-by: DrSmugleaf <DrSmugleaf@users.noreply.github.com>
Co-authored-by: Skarletto <122584947+Skarletto@users.noreply.github.com>
Co-authored-by: NULL882 <104377798+NULL882@users.noreply.github.com>
Co-authored-by: lunarcomets <140772713+lunarcomets@users.noreply.github.com>
Co-authored-by: Miro Kavaliou <miraslauk@gmail.com>
Co-authored-by: Kara <lunarautomaton6@gmail.com>
Co-authored-by: Kacper Urbańczyk <kacperjaroslawurbanczyk@gmail.com>
Co-authored-by: TsjipTsjip <19798667+TsjipTsjip@users.noreply.github.com>
Co-authored-by: daerSeebaer <61566539+daerSeebaer@users.noreply.github.com>
Co-authored-by: Fluffiest Floofers <thebluewulf@gmail.com>
Co-authored-by: Flareguy <78941145+Flareguy@users.noreply.github.com>
Co-authored-by: Leon Friedrich <60421075+ElectroJr@users.noreply.github.com>
Co-authored-by: Kevin Zheng <kevinz5000@gmail.com>
Co-authored-by: drteaspoon420 <87363733+drteaspoon420@users.noreply.github.com>
Co-authored-by: Ubaser <134914314+UbaserB@users.noreply.github.com>
Co-authored-by: Nim <128169402+Nimfar11@users.noreply.github.com>
Co-authored-by: Kacper Urbańczyk <mikrel071204@gmail.com>
Co-authored-by: Tox Cruize <141375638+TexCruize@users.noreply.github.com>
Co-authored-by: Ed <96445749+TheShuEd@users.noreply.github.com>
Co-authored-by: Michael Cu <43478115+michaelcu@users.noreply.github.com>

FluffiestFloof committed:
hol (#95)

FluffiestFloof committed:
Insulative Gear research and items (#93)

* Head

* change icon

* Not necessary

* Buh

Commit:
Automatic Changelog Update (#93)

DebugOk committed:
Prevent /suicide (#84)

Rebinds as /ghost instead

FluffiestFloof committed:
.38 special and revolvers (#92)

* Port all the things

* Fix all the things

* Bluespace doesn't exist apparently

* Add missing lathe recipes

* Not in this PR's scope

* Not necessary for this

Commit:
Automatic Changelog Update (#92)

Adrian16199 committed:
Near-sighted trait from SIMPLE STATION 14 (#73)

* Near-sighted trait from SIMPLE STATION 14

AKA PARKSTATION
COMING NOW TO THE REBASE!!!!
FUNNY CODE MADE BY DEATH (allegedly)[joke]{unless?}

After i figure out what does a component not have a setter mean.

* Aparently this was needed

The trait doesnt appear in the traits so thats funny.

* The - update

Aparently its very important.

---------

Signed-off-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>
Co-authored-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>

FluffiestFloof committed:
Tips (#81)

Commit:
Automatic Changelog Update (#81)

DebugOk committed:
Update changelog system (#75)

FluffiestFloof committed:
Nyanotrasen epistemic lore book (#82)

* The book

* locale the book.

* uncomment mail about that book

* imo useless trailing space

* Add book to spawner

* fix

Commit:
Automatic Changelog Update (#82)

FluffiestFloof committed:
Kanabou + Samurai crate (#83)

* Kanabou

* samurai crate

Commit:
Automatic Changelog Update (#83)

DebugOk committed:
Add hologram shader (#68)

* Add holograms!

* Updated localization files

Commit:
Automatic Changelog Update (#68)

PHCodes committed:
Reverse Engineering. (#91)

* Reverse Engineering.

It's engineering. It's reversed. I implemented Bluespace Crystals to get everything to work. I may have missed some recipes in the Lathes. Otherwise, everything seems like it just works. Blukie suits baby.

* Fixing the Bonus mechanic out of testing.

* Fixing Attributions.

* Fixing Attributions again because it won't let me put in multiple links into one string.

* Not dealing with Techfabs right now, integration tests will not pass with them.

* Added appropriate comments

---------

Co-authored-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>

FluffiestFloof committed:
Martial Artist PDA (#89)

Signed-off-by: Fluffiest Floofers <thebluewulf@gmail.com>

Commit:
Automatic Changelog Update (#89)

FluffiestFloof committed:
Bluespace is real (#96)

FluffiestFloof committed:
One book mail but three books (#97)

FluffiestFloof committed:
fix missing prototype (#98)
